lib/bolt/shell/powershell.rb

```diff
@@ -0,0 +1,263 @@
+                                          "to run it please add to 'winrm: extensions'", 'FILETYPE_ERROR')
+        end
+      end
+
+      def process_from_extension(path)
+        case Pathname(path).extname.downcase
+        when '.rb'
+          [
+            'ruby.exe',
+            %W[-S "#{path}"]
+          ]
+        when '.ps1'
+          [
+            'powershell.exe',
+            [*PS_ARGS, path]
+          ]
+        when '.pp'
+          [
+            'puppet.bat',
+            %W[apply "#{path}"]
+          ]
+        else
+          # Run the script via cmd, letting Windows extension handling determine how
+          [
+            'cmd.exe',
+            %W[/c "#{path}"]
+          ]
+        end
+      end
+
+      def escape_arguments(arguments)
+        arguments.map do |arg|
+          if arg =~ / /
+            "\"#{arg}\""
+          else
+            arg
+          end
+        end
+      end
+
+      def set_env(arg, val)
+        "[Environment]::SetEnvironmentVariable('#{arg}', @'\n#{val}\n'@)"
+      end
+
+      def quote_string(string)
+        "'" + string.gsub("'", "''") + "'"
+      end
+
+      def write_executable(dir, file, filename = nil)
+        filename ||= File.basename(file)
+        validate_extensions(File.extname(filename))
+        remote_path = "#{dir}\\#{filename}"
+        conn.copy_file(file, remote_path)
+        remote_path
+      end
+
+      def execute_process(path, arguments, stdin = nil)
+        quoted_args = arguments.map { |arg| quote_string(arg) }.join(' ')
+
+        quoted_path = if path =~ /^'.*'$/ || path =~ /^".*"$/
+                        path
+                      else
+                        quote_string(path)
+                      end
+        exec_cmd =
+          if stdin.nil?
+            "& #{quoted_path} #{quoted_args}"
+          else
+            <<~STR
+            $command_stdin = @'
+            #{stdin}
+            '@
+
+            $command_stdin | & #{quoted_path} #{quoted_args}
+            STR
+          end
+        Snippets.execute_process(exec_cmd)
+      end
+
+      def mkdirs(dirs)
+        mkdir_command = "mkdir -Force #{dirs.uniq.sort.join(',')}"
+        result = execute(mkdir_command)
+        if result.exit_code != 0
+          message = "Could not create directories: #{result.stderr.string}"
+          raise Bolt::Node::FileError.new(message, 'MKDIR_ERROR')
+        end
+      end
+
+      def make_tempdir
+        find_parent = target.options['tmpdir'] ? "\"#{target.options['tmpdir']}\"" : '[System.IO.Path]::GetTempPath()'
+        result = execute(Snippets.make_tempdir(find_parent))
+        if result.exit_code != 0
+          raise Bolt::Node::FileError.new("Could not make tempdir: #{result.stderr.string}", 'TEMPDIR_ERROR')
+        end
+        result.stdout.string.chomp
+      end
+
+      def rmdir(dir)
+        execute(Snippets.rmdir(dir))
+      end
+
+      def with_remote_tempdir
```

Comment:
This is `with_tempdir` in the bash shell (I *think* this is also used for local windows?)

Reply:
Good catch. I tried to standardize some of the method names but missed it.